tests/boardfarm_plugins/boardfarm_prplmesh/tests/initial_ap_config.py

```diff
@@ -0,0 +1,68 @@
+"""
+SPDX-License-Identifier: BSD-2-Clause-Patent
+SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+This code is subject to the terms of the BSD+Patent license.
+See LICENSE file for more details.
+"""
+
+import boardfarm
+
+from boardfarm.tests import bft_base_test
+from environment import sniffer
+
+
+class InitialApConfig(bft_base_test.BftBaseTest):
+    '''
+    Check initial configuration on device
+    '''
+
+    def runTest(self):
+        regex_list = [
+            r"WSC Global authentication success",
+            r"KWA \(Key Wrap Auth\) success",
+            r".* Controller configuration \(WSC M2 Encrypted Settings\)"
+        ]
+        failed_regex = []
+
+        wired_sniffer = sniffer.Sniffer(self.dev.board.gw_iface, True, boardfarm.config.output_dir)
+        wired_sniffer.start(self.__class__.__name__)
+
+        for radio in self.dev.board.al_entity.radios:
+            for regex in regex_list:
+                result, line, match = radio.wait_for_log(regex, 0, 0.3)
+                if not result:
+                    failed_regex.append((radio.iface_name, regex))
+
+        wired_sniffer.stop()
+
+        if failed_regex:
+            msg = ""
+            for entry in failed_regex:
+                msg += "Check for {}:{}: '{}' failed!\n".format(self.dev.board.al_entity.name,
+                                                                entry[0],
+                                                                entry[1]
+                                                                )
+            raise Exception(msg)
```

Comment:
Is raising an exception the only way to indicate failure?

Reply:
> Is raising an exception the only way to indicate failure?

Yes, `runTest()` methods doesn't have any return statement, so, according to boardfarm sources, the only way to indicate failure - throw exception.